### PR TITLE
Data Access Pane Preferences IO Refactor

### PR DESCRIPTION
**Pull request by formalhaunt**
### Description of the Change

This is a follow on from the PR https://github.com/constellation-app/constellation/pull/1356.

It is now utilising the newer version of the IO code allowing us to pass it an object rather than a Jackson JSON Node. This has allowed for the clean up of a significant amount of complicated code involving several loops. It has made it easier to test and hopefully easier to understand.

I also deleted a tonne of commented out code. It hasn't been touched for years so I figure if there was an issue it would have been fixed by now.

### Why Should This Be In Core?

Improves test coverage and readability.

### Benefits

Improves test coverage and readability.

### Verification Process

Wrote unit test before and verified still worked after. One note is that in the load preferences. The `<plugin>.__is_enabled__` property was always `null` after the load. This was due to the way the JSON de-serialisation was happening. The new code will actually set that value. Looking at the save preferences code it probably wasn't an issue as that code used the actual activation flag in the object to add this property rather than looking at the properties themselves.

### Applicable Issues

N/A

### CODE REVIEW

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/api/DataAccessUserPreferences.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+                    }
+                
+                });
+    }
+    
+    public Map<String, String> getGlobalParameters() {
+        return globalParameters;
+    }
+
+    public void setGlobalParameters(Map<String, String> globalParameters) {
```
Comment (antares1470) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/api/DataAccessUserPreferences.java:109]: parameter final

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/api/DataAccessUserPreferences.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        this.globalParameters = globalParameters;
+    }
+
+    public Map<String, String> getPluginParameters() {
+        return pluginParameters;
+    }
+
+    public void setPluginParameters(Map<String, String> pluginParameters) {
```
Comment (antares1470) [inline CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/api/DataAccessUserPreferences.java:117]: same here

**Review by antares1470 — CHANGES_REQUESTED** [review]
Looks like there are some errors according to github actions.

Side note: Don't know if it might be worth chatting to @mimosa2 as they also have a PR up relating to `ParameterIOUtilities.loadParameters()` (#1390)

File: CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/api/DataAccessUserPreferences.java (hunk 1 above)
Comment (formalhaunt) [reply CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/api/DataAccessUserPreferences.java]: Fixed

**Comment by formalhaunt** [comment]
Build is fixed and I have pulled in @mimosa2 changes

**Review by antares1470 — CHANGES_REQUESTED** [review]
I have an exception for you:
```
java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at java.base/java.util.ImmutableCollections$Map1.<init>(ImmutableCollections.java:749)
	at java.base/java.util.Map.of(Map.java:1307)
	at au.gov.asd.tac.constellation.views.dataaccess.api.DataAccessUserPreferences.lambda$toPerPluginParamMap$6(DataAccessUserPreferences.java:157)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at au.gov.asd.tac.constellation.views.dataaccess.api.DataAccessUserPreferences.toPerPluginParamMap(DataAccessUserPreferences.java:162)
	at au.gov.asd.tac.constellation.views.dataaccess.io.ParameterIOUtilities.lambda$loadParameters$5(ParameterIOUtilities.java:208)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at au.gov.asd.tac.constellation.views.dataaccess.io.ParameterIOUtilities.loadParameters(ParameterIOUtilities.java:193)
	at au.gov.asd.tac.constellation.views.dataaccess.panes.DataAccessPane.lambda$new$9(DataAccessPane.java:342)
```
^ this happened when I tried a load a template which merely had the remove nodes and merge nodes plugins ticked without any changes made to the parameters. master didn't have such troubles loading it.

I think there may also be something wrong with the save templates. When I saved the aforementioned template in your branch and then loaded it in master, neither of those plugins were selected. Didn't encounter such issues when I saved the same template in master

**Comment by formalhaunt** [comment]
The null pointer should now be resolved. Verified against your test cases and added in tests to cover this edge case.

**Review by antares1470 — APPROVED** [review]
All the previous issues appear to be resolved :)

**Comment by arcturus2** [comment]
I've found another NPE @formalhaunt but that exists in 2.4 so not your issue. I won't hold your PR up any longer and will put through a small PR to fix that. 

Thanks for this fix.

**Review by arcturus2 — APPROVED** [review]
Thanks for this.
